documentation/src/main/asciidoc/server_guide/configuration.adoc

```diff
@@ -499,49 +499,45 @@ By default each protocol stores data in the cache in the most efficient format f
 
 ----
 
-To specify a custom server-side compatibility marshaller use the "marshaller" attribute:
+To specify a custom server-side compatibility marshaller use the "marshaller" attribute which can hold either an
+ordinary fully qualified class name or an 'extended' class name which is composed of a JBoss Modules module identifier,
+a slot name, and a fully qualified class name separated by the ':' character, (eg. "com.acme.my-module-name:my-slot:com.acme.CustomMarshaller").
 
-[source,xml]
-----
-
-<compatibility marshaller="com.acme.CustomMarshaller"/>
+By using a plain fully qualified class name you will be able to reference only 'stock' marshallers provided by Infinispan,
+like "org.infinispan.query.remote.CompatibilityProtoStreamMarshaller". This option is provided for conciseness and backwards
+compatibility. The second option of using 'extended' class names is more generic and is preferrd as it allows you to
+really add new marshallers from an external module of your choice.
 
-----
+The module identifier and slot follow the rules imposed by JBoss Modules, and the designated module must be available to
+the server either in the 'modules' folder or be deployed through the deployer service in the 'deployments' folder. In
+the second case its module identifier will be of the form 'deployment.my-jar-name.jar' and the slot will be 'main'.
 
-Your custom marshaller needs to be on the classpath of the Infinispan module. You can add it by either:
+NOTE: The referenced module will become an automatic dependency for the Cache referencing it, so in case the jar
+defining the module is redeployed, the Cache will be restarted.
```

Comment:
In theory everything can be controlled, as long as the handlers can apply the change at runtime.
For dependencies however this is tricky, since I imagine that a deployment is actually a removal and addition and the in-between situation would be undefined.